extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/core/GreenbidsInferenceDataService.java

```diff
@@ -86,23 +88,34 @@ private List<ThrottlingMessage> extractMessagesForImp(
         final String ip = Optional.ofNullable(bidRequest.getDevice())
                 .map(Device::getIp)
                 .orElse(null);
-        final String countryFromIp = getCountry(ip);
+        final String country = Optional.ofNullable(bidRequest.getDevice())
+                .map(Device::getGeo)
+                .map(Geo::getCountry)
+                .map(GreenbidsInferenceDataService::getCountryNameFromAlpha3)
+                .orElseGet(() -> getCountry(ip));
+
```

Comment:
this change is not covered by the unit tests

Reply:
After checking [Locale docs](https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#getCountry--), the alpha3 code is not convertable to countryName via `getCountry`. 
<img width="1124" alt="image" src="https://github.com/user-attachments/assets/e66fedee-bf10-418f-8b98-921cb3b6fc74" />

So need to use CountryCodeMapper] [mapToAlpha2](https://github.com/EvgeniiMunin/prebid-server-java/blob/f31ec919bdbd08632bd0723116f4c5fde553d3cd/src/main/java/org/prebid/server/geolocation/CountryCodeMapper.java#L30) to convert alpha3 -> alpha2 and then via Locale alpha2 -> countryName.

Reply:
sure, if you need it

Reply:
Added CountryCodeMapper for alpha3 -> alpha2 conversion